pkg/gardenlet/controller/shoot/care/garbage_collection.go

```diff
@@ -117,25 +118,29 @@ func (g *GarbageCollection) deleteOrphanedNodeLeases(ctx context.Context, c clie
 		return err
 	}
 
-	var orphanedLeases []client.Object
+	var taskFns []flow.TaskFn
 
 	for _, l := range leaseList.Items {
 		if len(l.OwnerReferences) > 0 {
 			continue
 		}
 		lease := l.DeepCopy()
 
-		if err := c.Get(ctx, client.ObjectKey{Name: lease.Name}, &metav1.PartialObjectMetadata{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Node"}}); err != nil {
-			if !apierrors.IsNotFound(err) {
-				return fmt.Errorf("failed getting node %s when checking for potential orphaned Lease %s: %w", lease.Name, client.ObjectKeyFromObject(lease), err)
+		taskFns = append(taskFns, func(ctx context.Context) error {
+			if err := c.Get(ctx, client.ObjectKey{Name: lease.Name}, &metav1.PartialObjectMetadata{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Node"}}); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return fmt.Errorf("failed getting node %s when checking for potential orphaned Lease %s: %w", lease.Name, client.ObjectKeyFromObject(lease), err)
+				}
+
+				g.log.Info("Detected orphaned Lease object, cleaning it up", "nodeName", lease.Name, "lease", client.ObjectKeyFromObject(lease))
+				return kubernetesutils.DeleteObject(ctx, c, lease)
```

Comment:
Shouldn't this be outside the `If`? Also return nil if the object is already gone.